src/layer/poi.js

```diff
@@ -0,0 +1,103 @@
+import * as label from "../constants/label.js";
+import * as Color from "../constants/color.js";
+
+var iconDefs = {
+  bar: ["bar", "beer"],
+  hospital: "hospital",
+  medical: ["doctors", "clinic"],
+  parking: "parking",
+};
+
+export const poi = {
+  id: "poi",
+  type: "symbol",
+  paint: {
+    "text-halo-color": Color.backgroundFill,
+    "text-halo-width": 1.5,
+    "icon-halo-width": 0.4,
+    "text-halo-blur": 1,
+    "icon-halo-blur": 0.2,
+    "text-color": [
+      "match",
+      ["get", "class"],
+      ["bar", "beer"],
+      Color.poi.consumer,
+      ["hospital", "parking"],
+      Color.poi.infrastructure,
+      Color.poi.infrastructure,
+    ],
+  },
+  filter: [
+    ">=",
+    ["zoom"],
+    [
+      "match",
+      ["get", "subclass"],
```

Comment:
The `text-color` property matches on the feature’s `class` property, while the filter matches on the feature’s `subclass` property, listing out a similar set of values. Can we stick to one of these properties, or do we need the extra specificity of `subclass`?

Reply:
For background, when a POI class doesn't have `subclass`, `subclass` just repeats what's in `class`.  I haven't studied the POI structure comprehensively enough to know if it's actually sensible to apply styling at the `class` level versus `subclass`.  We'll have the most flexibility if we go all-in on `subclass` but that might come at the expense of cases where we can group an entire `class` together with common styling.

Reply:
If we’re mostly concerned with cases where `subclass` is missing, then a `coalesce` expression could split the difference.